app/helpers/tasks_helper.rb

```diff
@@ -0,0 +1,46 @@
+module TasksHelper
+  TASKS_LIMIT = 5
+
+  def all_assigned_tasks
+    @assigned_tasks ||= begin
+      tasks = current_user.cards
+        # Using Arel.sql to sort the records by due_date with null due_date records last
+        .order(Arel.sql('due_date IS NULL, due_date ASC'))
+
+      if params[:project_id]
+        return tasks.select { |card| card.project.id == current_project.id }
+      end
+
+      tasks
+    end
+  end
+
+  def assigned_tasks
+    all_assigned_tasks.first(TASKS_LIMIT)
+  end
+
+  def overflow_tasks_count
```

Comment:
#suggestion 

`overflow_tasks_count` leads you to expect a number, so what about returning `0` on L23 and checking for > 0 in the view?